galleries/examples/specialty_plots/ishikawa_diagram.py

```diff
@@ -108,82 +109,80 @@ def causes(data: list, cause_x: float, cause_y: float,
 
 def draw_body(data: dict):
     """
-    Place each section in its correct place by changing
+    Place each problem section in its correct place by changing
     the coordinates on each loop.
 
     Parameters
     ----------
     data : dict
-        The input data (can be list or tuple). ValueError is
-        raised if more than six arguments are passed.
+        The input data (can be a dict of lists or tuples). ValueError
+        is raised if more than six arguments are passed.
 
     Returns
     -------
     None.
 
     """
-    second_sections = []
-    third_sections = []
-    # Resize diagram to automatically scale in response to the number
-    # of problems in the input data.
-    if len(data) == 1 or len(data) == 2:
-        spine_length = (-2.1, 2)
-        head_pos = (2, 0)
-        tail_pos = ((-2.8, 0.8), (-2.8, -0.8), (-2.0, -0.01))
-        first_section = [1.6, 0.8]
-    elif len(data) == 3 or len(data) == 4:
-        spine_length = (-3.1, 3)
-        head_pos = (3, 0)
-        tail_pos = ((-3.8, 0.8), (-3.8, -0.8), (-3.0, -0.01))
-        first_section = [2.6, 1.8]
-        second_sections = [-0.4, -1.2]
-    else:  # len(data) == 5 or 6
-        spine_length = (-4.1, 4)
-        head_pos = (4, 0)
-        tail_pos = ((-4.8, 0.8), (-4.8, -0.8), (-4.0, -0.01))
-        first_section = [3.5, 2.7]
-        second_sections = [1, 0.2]
-        third_sections = [-1.5, -2.3]
-
-    # Change the coordinates of the annotations on each loop.
+    # Set the length of the spine according to the number of 'problem' categories.
+    length = 0
+    for i in range(math.ceil(len(data) / 2)):
+        length = i
```

Comment:
Should this be

```suggestion
        length += i
```
?

Reply:
It's correct as it is. `length` can be either 0, 1 or 2 depending on how many categories there are.
It'll keep looping until it lands on the right one and then ends the loop.